api/tasks_test.go

```diff
@@ -59,12 +59,45 @@ func expectedTask(cfg *config.Config, jobID, taskName string, jsonResponse bool,
 	if jsonResponse {
 		task.Links.Job = fmt.Sprintf("%s/%s%s", cfg.BindAddr, cfg.LatestVersion, task.Links.Job)
 		task.Links.Self = fmt.Sprintf("%s/%s%s", cfg.BindAddr, cfg.LatestVersion, task.Links.Self)
+	} else {
+		taskETag, err := models.GenerateETagForTask(ctx, task)
+		if err != nil {
+			t.Errorf("failed to generate eTag for expected test job - error: %v", err)
```

Comment:
Why does the GenerateETagForTask function only get called if the jsonResponse is false? Perhaps a comment to explain this would be useful here. I notice that in the expectedJob function it gets called whatever the jsonResponse is: https://github.com/ONSdigital/dp-search-reindex-api/blob/feature/etag-get-jobs-tasks-id/api/jobs_test.go#:~:text=%3A%3D%20models.-,GenerateETagForJob,-(ctx%2C 

Also a small typ here:  
```suggestion
			t.Errorf("failed to generate eTag for expected test task - error: %v", err)
```